libgloss/arc/hl/hl_argv.c

```diff
@@ -0,0 +1,50 @@
+/*********************************************************************
+Copyright  2020 Synopsys Inc.
+*********************************************************************/
+
+#include <stddef.h>
+
+#include "hl_toolchain.h"
+#include "hl_api.h"
+
+static uint32_t
+hl_argvlen (int a)
+{
+  uint32_t ret = 0;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_ARGV, "i", (uint32_t) a);
+  if (p != NULL)
+    ret = hl_get_ptr_len (p);
+
+  hl_delete ();
+
+  return ret;
+}
+
+static int
+hl_argv (int a, char *arg)
+{
+  int ret = 0;
+  volatile __uncached char *p;
+
+  p = hl_message (HL_SYSCALL_ARGV, "i:s", (uint32_t) a, (char *) arg);
+  if (p == NULL)
+    ret = -1;
+
+  hl_delete ();
+
+  return ret;
+}
+
+int
+_argv (int a, char *arg)
```

Comment:
Same function like above, just drop this one, and use the above one as global

Reply:
I want to have it because it makes code clearer and more predictable for the reader. In my opinion.

Reply:
Can be, but the file is soo small, that it make no real sense. Alternative, please use always inline construct to inline the static code into the global call.

Reply:
Added always_inline everywhere, thanks!